src/server/controllers/webhook-events.ts

```diff
@@ -21,8 +26,76 @@ type PullRequestReviewHook = {
   repository: RepositoryHook;
 };
 
+type StatusHook = {
+  sha: string;
+  name: string;
+  state: 'error'|'failure'|'pending'|'success'; description: string;
+  repository: RepositoryHook;
+  commit: {author: User;}
+};
+
 // Triggered when the status of a Git commit changes.
-export async function handleStatus(_hookBody: string) {
+export async function handleStatus(hookData: StatusHook) {
+  if (hookData.state !== 'error' && hookData.state !== 'failure') {
+    return;
```

Comment:
It might be useful from a debugging perspective if we returned different 2xx codes depending on what we do. eg. 202 on cases where we don't do anything with the webhook event.

(Don't need to address in this PR).